### DISCUSSION TITLE
Question regarding mismatch in query parameters listed in documentation and in code for image push and pull libpod API

### DISCUSSION THREAD

**Discussion by tarun360** [containers/podman#16551] [Q&A]
In the image push libpod API ([link](https://docs.podman.io/en/latest/_static/api.html#tag/images/operation/ImagePushLibpod)), only three query parameters are listed - `destination`, `quiet` and `tlsVerify`. Whereas in the code [here](https://github.com/containers/podman/blob/da6d57e13d702f3bac202e70aacc59ebe261050e/pkg/api/handlers/libpod/images_push.go#L27) I see more query parameters, viz., `CompressionFormat`, `Format`, `All`, `RemoveSignatures`. Why is there a mismatch in the query parameters in the code and that listed in documentation? Also in the [comment](https://github.com/containers/podman/blob/da6d57e13d702f3bac202e70aacc59ebe261050e/pkg/api/handlers/libpod/images_push.go#L22) just above the `PushImage` function it says it is the compat http endpoint, isn't this the libpod endpoint?

For image pull libpod API, there's a `credentials` query parameter mentioned in the [documentation](https://docs.podman.io/en/latest/_static/api.html#tag/images/operation/ImagePullLibpod), but that query parameter is not present in the code ([link](https://github.com/containers/podman/blob/da6d57e13d702f3bac202e70aacc59ebe261050e/pkg/api/handlers/libpod/images_pull.go#L30)). 

Could someone explain the mismatch in the query parameters listed in the code and that listed in the documentation?

**Comment by Luap99** [comment]
The is no real answer here. We have no automated testing to check swagger against the the real code like we have for the man pages.

So it is easy for the swagger to get out of date if reviewers miss such changes (and this is very is to do because the swagger is often defined in a different file).

So I suggest you open issues if you find a mismatch and they will be fixed eventually. And even better, if you know hot to fix the issues, open PRs for them.
